core/db/migrate/20160225152313_remove_tax_rate_from_shipping_rate.rb

```diff
@@ -0,0 +1,29 @@
+class RemoveTaxRateFromShippingRate < ActiveRecord::Migration
+  def up
+    Spree::ShippingRate.find_each do |shipping_rate|
```

Comment:
@adammathys you are totally right. 

But in this case we need a trusted interface and S.P.O.T. of `TaxRate#compute_amount`. 
That interface will not change for a long time (we can ensure that, because we own it). 
The internal implementation will change, though. But that's ok.

If we ensure that the public interface will not change, then violating the rules here is safe IMO.